### ISSUE TITLE
Abort called from: ecl_smspec_fwrite Bug (2018-10)

### ISSUE THREAD

**Issue by OPMUSER** [OPM/opm-simulators#1653]
If the case name is mixed case (Siminput.data) then the model runs until it attempts the first write to the SMSPEC file and aborts with:
`
Abort called from: ecl_smspec_fwrite (/build/ecl-5pxN8C/ecl-2018.10-rfinal/lib/ecl/ecl_smspec.cpp:496) 

Error message: ecl_smspec_fwrite: Unable to open fortio file (null), error: Bad address .
--------------------------------------------------------------------------------
 #00 util_abort__(..)                                                              in ???
 #01 ecl_smspec_fwrite(..)                                                         in ???
 #02 ecl_sum_fwrite(..)                                                            in ???
 #03 _ZN3Opm9EclipseIO13writeTimeStepEibdNS_12RestartValueERKSt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEdSt4lessIS8_ESaISt4pairIKS8_dEEERKS2_IS8_St6vectorIdSaIdEESA_SaISB_ISC_SK_EEERKS2_ISB_IS8_iEdS9_ISQ_ESaISB_IKSQ_dEEEb(..)    in ???
 #04 _ZN5Ewoms9EclWriterINS_10Properties4TTag14EclFlowProblemEE15EclWriteTasklet3runEv(..)    in ???
 #05 _ZN5Ewoms13TaskletRunner18startWorkerThread_EPS0_i(..)                        in ???
 #06 ???? 
 #07 ???(..)                                                                       in /build/glibc-OTsEL5/glibc-2.27/nptl/pthread_create.c:463
 #08 clone(..)                                                                     in /build/glibc-OTsEL5/glibc-2.27/misc/../sysdeps/unix/sysv/linux/x86_64/clone.S:97
--------------------------------------------------------------------------------
`
Changing the case name to upper case resolves the issue.

**Comment by OPMUSER** [comment]
Apologizes for the test layout, for some unknown reason I cannot set the text to code, and I cannot Preview either at the moment

**Comment by joakim-hove** [comment]
> If the case name is mixed case (Siminput.data) then the model runs until it attempts the first write to the SMSPEC file and aborts with:

Yes - this is unfortunate. This is to follow the eclipse lead - which also does not allow miXeD case basenames. The problem is the situation where filenames are generated based on a rule/convention and an a basename. I.e. if your data file is called `my_case.data` you will get the restart file `my_case.unrst`, and if your data file is is called `MY_CASE.DATA` you will get the restart file `MY_CASE.UNRST` - but how about `My_Case.Data`?

As I see it there are two options:

1. `flow` can detect MixeD case in the basename already on startup and refuse to continue.
2. We can establish a new convention  i.e. MiXed case will always give a fully capitalized extension - as `MiXeD.UNRST`. I don't know how that will play with other third-party tools.

Opinions and suggestions welcome.

NB: This is is my interpretation of the Eclipse behaviour - if someone thinks I have got it wrong I am happy to corrected.

**Comment by OPMUSER** [comment]
Eclipse runs mixed case on Windows as expected, just to make things even more confusing. So this was difficult to find.

I would suggest we keep this simple, and just convert to upper (or lower) case and continue, maybe with a message saying that case conversion was applied.

**Comment by joakim-hove** [comment]
Ok - but what will be the case on the extension if you mixed case in the basename?

**Comment by OPMUSER** [comment]
I don't think this really matters as long as the job runs and the results can be loaded into ResInsight.

I'll test this tomorrow to see if ResInsight loads various combinations on both Linux and Windows and will let you know. Whatever works with ResInsight will be fine.

**Comment by OPMUSER** [comment]
Okay, ResInsight has the same problem both basename and the extension must both be all uppercase or all lowercase. And this is for both Linux and Windows versions of ResInsight.

So to avoid issues with ResInsight  we should go with:

(1) flow can detect MixeD case in the basename already on startup and refuse to continue, but with a nice message that none C++ programmers can understand ;-)

At least then we don't cause any confusion with ResInsight not loading OPM Flow generated files.

**Comment by joakim-hove** [comment]
Ok sorry - I should have said yesterday, Resinsight uses the same code, so that will always behave in the same way.

What I was more concerned with was *other* third-party applications like S3Graf and Petrel.

**Comment by OPMUSER** [comment]
If we use option (1) then it will work with third party applications as normal, so no worries.
